src/typing/typer.ml

```diff
@@ -36,6 +36,12 @@ open Operators
 (* ---------------------------------------------------------------------- *)
 (* TOOLS *)
 
+let mono_or_dynamic ctx with_type p = match with_type with
+	| WithType.NoValue ->
+		t_dynamic
```

Comment:
should it be Void maybe?

Reply:
I'm a bit worried about some crazy inline situation where we end up with Void in a value-place... With Dynamic we at least have the same behavior as before.